ios/FluentUI/Core/Colors.swift

```diff
@@ -281,67 +281,151 @@ public final class Colors: NSObject {
+    @objc public static var primary: UIColor {
+        get {
+            return FallbackThemeColor.primary
+        }
+        set {
+            FallbackThemeColor.primary = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "setProvider(_:forWindow:)")
-    @objc public static var primaryTint10: UIColor = Palette.communicationBlueTint10.color
+    @objc public static var primaryTint10: UIColor {
+        get {
+            return FallbackThemeColor.primaryTint10
+        }
+        set {
+            FallbackThemeColor.primaryTint10 = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "setProvider(_:forWindow:)")
-    @objc public static var primaryTint20: UIColor = Palette.communicationBlueTint20.color
+    @objc public static var primaryTint20: UIColor {
+        get {
+            return FallbackThemeColor.primaryTint20
+        }
+        set {
+            FallbackThemeColor.primaryTint20 = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "setProvider(_:forWindow:)")
-    @objc public static var primaryTint30: UIColor = Palette.communicationBlueTint30.color
+    @objc public static var primaryTint30: UIColor {
+        get {
+            return FallbackThemeColor.primaryTint30
+        }
+        set {
+            FallbackThemeColor.primaryTint30 = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "setProvider(_:forWindow:)")
-    @objc public static var primaryTint40: UIColor = Palette.communicationBlueTint40.color
+    @objc public static var primaryTint40: UIColor {
+        get {
+            return FallbackThemeColor.primaryTint40
+        }
+        set {
+            FallbackThemeColor.primaryTint40 = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "setProvider(_:forWindow:)")
-    @objc public static var primaryShade10: UIColor = Palette.communicationBlueShade10.color
+    @objc public static var primaryShade10: UIColor {
+        get {
+            return FallbackThemeColor.primaryShade10
+        }
+        set {
+            FallbackThemeColor.primaryShade10 = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "setProvider(_:forWindow:)")
-    @objc public static var primaryShade20: UIColor = Palette.communicationBlueShade20.color
+    @objc public static var primaryShade20: UIColor {
+        get {
+            return FallbackThemeColor.primaryShade20
+        }
+        set {
+            FallbackThemeColor.primaryShade20 = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "setProvider(_:forWindow:)")
-    @objc public static var primaryShade30: UIColor = Palette.communicationBlueShade30.color
+    @objc public static var primaryShade30: UIColor {
+        get {
+            return FallbackThemeColor.primaryShade30
+        }
+        set {
+            FallbackThemeColor.primaryShade30 = newValue
+        }
+    }
 
     @available(*, deprecated, renamed: "textOnAccent")
```

Comment:
why does this have to change?

Reply:
Since you mentioned earlier that clients may change `Colors.primary` which also affects fallback colors in `Colors.primary(for:)`, I just move the actual storage to `FallbackThemeColor.primay`. Now both `Colors.primary(for:)` and `Colors.primary` refer to the same fallback color.

Reply:
github selected the wrong thing. I meant for foregroundprimary. it doesn't have to do with branding color

Reply:
Okay my intention is to keep the logic consistent (as with other colors). I can revert to the original logic for foregroundprimary.